tools/taskcluster/tasks.py

```diff
@@ -0,0 +1,143 @@
+        git config advice.detachedHead false
+        git checkout {self.commit}
+        yes | sdkmanager --licenses
+        ./gradlew -PisPullRequest clean assembleSystem assembleAndroidTest lint checkstyle ktlint pmd detekt test
+        ./gradlew -Pcoverage jacocoSystemDebugTestReport
+        ./tools/taskcluster/upload-coverage-report.sh
+        ./tools/taskcluster/download-firebase-sdk.sh
+        ./tools/taskcluster/google-firebase-testlab-login.sh
+        ./tools/taskcluster/execute-firebase-test.sh system/debug app-system-debug model=sailfish,version=25,orientation=landscape
+        '''
+
+        return self._craft_shell_task(
+            "Firefox for Amazon's Fire TV - Build - Pull Request",
+            script,
+            ['secrets:get:project/mobile/firefox-tv/tokens'],
+            {
+                'public/reports': artifact('directory', '/opt/firefox-tv/app/builds/reports')
+            }
+        )
+
+    def craft_master_task(self):
+        script = f'''
+        git fetch {self.repo_url}
+        git config advice.detachedHead false
+        git checkout {self.commit}
+        yes | sdkmanager --licenses
+        ./gradlew -PisPullRequest clean assembleSystem assembleAndroidTest lint checkstyle ktlint pmd detekt test
+        python ./tools/taskcluster/get-bitbar-token.py
+        python ./tools/taskcluster/execute-bitbar-test.py system/debug app-system-debug
+        '''
+
+        return self._craft_shell_task(
+            "Firefox for Amazon's Fire TV - Build - Master",
+            script,
+            ['secrets:get:project/mobile/firefox-tv/tokens'],
+            {
+                'public': artifact('directory', '/opt/firefox-tv/app/builds/reports')
+            }
+        )
+
+    def craft_release_task(self, tag):
+        script = f'''
+        git fetch origin --tags
+        git config advice.detachedHead false
+        git checkout {tag}
+        yes | sdkmanager --licenses
+        python tools/taskcluster/get-sentry-token.py
+        python tools/taskcluster/get-pocket-token.py
+        ./gradlew --no-daemon clean test assembleSystemRelease
+        '''
+
+        return self._craft_shell_task(
+            f'Firefox for Fire TV - Release build {tag}',
+            script,
+            ['secrets:get:project/mobile/firefox-tv/tokens'],
+            {
+                'public': artifact('directory', '/opt/firefox-tv/app/build/outputs/apk')
+            }
+        )
+
+    def _craft_shell_task(self, name, script, scopes, artifacts):
+        single_command = ' && '.join([line.strip() for line in script.split('\n') if line.strip()])
+        bash_command = [
+            '/bin/bash',
+            '--login',
+            '-cx',
+            f'export TERM=dumb && {single_command}'
+        ]
+
+        return {
+            'taskGroupId': self.task_group_id,
+            'provisionerId': 'aws-provisioner-v1',
+            'schedulerId': 'taskcluster-github',
+            'workerType': 'github-worker',
```

Comment:
Okay to use github-worker for the sake of this PR. Let's move it to `mobile-X-b-fftv` as soon as possible, since we're shipping product off these workers.